spec/support/shared_examples/webauthn_setup.rb

```diff
@@ -38,4 +38,54 @@
 
     expect_webauthn_setup_error
   end
+
+  context 'platform authenticator logging' do
+    before do
+      allow(IdentityConfig.store).
+        to receive(:show_unsupported_passkey_platform_authentication_setup).
+        and_return(true)
+      allow(WebauthnVerificationForm).to receive(:domain_name).and_return('localhost:3000')
+      allow_any_instance_of(ApplicationController).to receive(:analytics).and_return(fake_analytics)
+      mock_webauthn_setup_challenge
+    end
+
+    let!(:user) { sign_up_and_set_password }
+    let(:fake_analytics) { FakeAnalytics.new }
```

Comment:
Couple nitpicks:

- We're not using `user` for anything and I find that relying on `let!`s behavior of eagerly initializing a variable for all specs to be a little harder to follow vs. putting the equivalent behavior in `before`
- I would expect `let`s to come before the `before`, especially when the `before` references those variables

```suggestion
    let(:fake_analytics) { FakeAnalytics.new }

    before do
      allow(IdentityConfig.store).to receive(:platform_auth_set_up_enabled).and_return(true)
      allow(IdentityConfig.store).
        to receive(:show_unsupported_passkey_platform_authentication_setup).
        and_return(true)
      allow(WebauthnVerificationForm).to receive(:domain_name).and_return('localhost:3000')
      allow_any_instance_of(ApplicationController).to receive(:analytics).and_return(fake_analytics)
      sign_up_and_set_password
      mock_webauthn_setup_challenge
    end
```

Reply:
The setup of `user` in both `management_spec.rb` and `sign_up_spec.rb` were done differently in each file, so that `let!` seemed helpful in creating a standardized user. I am unsure of an alternative solution right now, but I am open to other ideas.